types.go

```diff
@@ -123,7 +151,7 @@ type Task struct {
 	UID        int64               `json:"uid,omitempty"`
 	TaskUID    int64               `json:"taskUid,omitempty"`
 	IndexUID   string              `json:"indexUid"`
-	Type       string              `json:"type"`
+	Type       TaskType            `json:"type"`
```

Comment:
Is this change can be considered a breaking change right?

Reply:
Good catch @brunoocasali, yes, absolutely.

Perhaps we can stick with `string` and make the type change when we rewrite for generic types, WYDT ? (https://github.com/meilisearch/meilisearch-go/issues/463)

Reply:
Since the SDK is not stable yet, is it a breaking we could afford for the sake of the library?
If we want to reach stability at some point we might need to accept this kind of change.
What do you think @42atomys and @brunoocasali?

Reply:
I agree 100% with you, as v0 is usually assumed to be "on the path to the stable version."

I looked through the types defined in the files, and this is the only strict string value I found (except for generics as well).

If we decide on it, we need to ensure clear release communication.

Reply:
@42atomys can you help me with the communication by updating your PR description with what the users had to do before and what they must do now (with examples like a few lines of code if possible)?
I will copy paste this to add it to the changelog under a "⚠️ Breaking changes" section. Plus, the minor of the version will be updated
So it should be enough regarding the communication

Reply:
I'm also okay with breaking changes here if they are transparent to the user about how to deal with them.